### PR TITLE
8.x 4.x dev

### PR DESCRIPTION
**Pull request by trent-crawford**

### CODE REVIEW

File: src/EventSubscriber/CommerceEventsSubscriber.php [hunk 1]
```diff
@@ -81,7 +83,11 @@ public function trackProductView(GetResponseEvent $event) {
       $default_variation = $product->getDefaultVariation();
 
       if ($default_variation) {
-        $this->eventTracker->productDetailViews([$default_variation]);
+        // This is event has been disabled as it causes a session to be opened
+        // when the user is browsing. This increases load on origin
```
Comment (grahl) [inline src/EventSubscriber/CommerceEventsSubscriber.php:87]: > This increases load on origin significantly.

Could be rephrased to a more generic behavior.

File: src/Product.php [hunk 2]
```diff
@@ -12,56 +14,56 @@ class Product {
    *
    * @var string
    */
-  private $name;
+  private string $name;
```
Comment (grahl) [inline src/Product.php:17]: Typed properties are available from 7.4 forward, the module does not enforce that though so compatibility with Core 8.9 means that it isn't guaranteed to be available.

File: src/Product.php [hunk 3]
```diff
@@ -73,11 +75,27 @@ public function toArray() {
     $data = [];
 
     foreach ($this as $property => $value) {
-      $property = ($property != 'price') ? 'item_' . $property : $property;
+      // Special cases for plural to singular conversion.
+      $property = $property === 'categories' ? 'category' : $property;
+
+      // Special case of price without item prefix.
+      $property = ($property !== 'price') ? 'item_' . $property : $property;
```
Comment (grahl) [inline src/Product.php:82]: Parentheses used inconsistently here compared to line 79, consider harmonizing them.

Comment (trent-crawford) [reply src/Product.php:82]: Thanks. I will standardize here.

File: src/Product.php (hunk 2 above)
Comment (trent-crawford) [reply src/Product.php:17]: Good catch. Thank you.

File: src/EventSubscriber/CommerceEventsSubscriber.php (hunk 1 above)
Comment (trent-crawford) [reply src/EventSubscriber/CommerceEventsSubscriber.php:87]: Good idea. Thank you.

**Pull request merged**